File: Algorithms/Easy/905_SortArrayByParity/solution.py
from typing import List


class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        l, r = 0, len(nums) - 1
        while l < r:
            if nums[l] % 2:
                while l < r and nums[r] % 2:
                    r -= 1
                nums[l], nums[r] = nums[r], nums[l]
            l += 1
        return nums

File: Algorithms/Easy/905_SortArrayByParity/Solution.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& nums) {
        vector<int> even, odd, ans;
        int n = nums.size();
        
        for (int i = 0; i < n; i++) {
            if (nums[i] & 1)
                odd.push_back(nums[i]);
            else
                even.push_back(nums[i]);
        }
        
        for (int i = 0; i < even.size(); ans.push_back(even[i]), i++);
        for (int i = 0; i < odd.size(); ans.push_back(odd[i]), i++);
        
        return ans;
    }
};